content/attacks/posts/2021-02-13-Alpha-Finance.md

```diff
@@ -0,0 +1,52 @@
+The team also [collaborated](https://twitter.com/stellaxyz_/status/1360535699368251394?s=20) with authorities and cybersecurity experts to [trace the stolen funds](https://www.quadrigainitiative.com/casestudy/alphahomorahack.php) and prevent the attacker from profiting. Efforts were made to freeze the remaining assets on cryptocurrency exchanges, aiming to safeguard the funds and mitigate the impact of the attack.
+
+- **February 20, 2021:**
+An agreement is reached between Alpha Homora V2 (Alpha Finance Lab) and CREAM V2 (CREAM) regarding the amount of funds and repayment mechanics. The funds in discussion include 13,245 ETH, 4,263,139 DAI, 4,032,014 USDC, and 5,647,242 USDT, along with the accrued interest. The borrowing interest rate on the funds is halted through an upgrade to the CREAM V2 contract. Alpha Finance Lab will cover the cost of auditing the proposed change. Additionally, $50 million worth of ALPHA tokens will be put in an escrow contract with a 7-day timelock, and they will be released back to Alpha Finance Lab periodically and proportionally as the funds are repaid.
+
+- **March 1, 2023:** 
+Iron Bank (IB) made a unilateral decision to modify the smart contract configuration, leading to the prevention of lenders on Alpha Homora (AH) from withdrawing their liquidity. This action was taken due to an unresolved debt between AH and IB, which originated from a code exploit attack in 2021. Despite AH's commitment to a repayment plan, IB failed to liquidate collateral as required, resulting in a vulnerability in their protocol. Without AH's knowledge, IB upgraded their code and froze depositors' funds, demanding a resolution.
+Over a period of 2.5 months, AH, on behalf of the depositors, engaged in negotiations with IB and proposed multiple solutions to unfreeze the funds. Despite their efforts, IB rejected all proposed solutions and maintained the freeze on the funds. Consequently, depositors unanimously voted to cease negotiations with IB, receive goodwill funds from AH, and initiate legal action against IB to unfreeze their funds.
+
+
+## Security Failure Causes
+
+- **Loophole in Custom Spells:** The Alpha Homora v2 contract allowed the usage of custom spells without strict checks, providing an avenue for the attacker to execute the attack.
+- **Rounding Error Exploitation:** The attacker took advantage of a rounding error in the protocol during repayment, resulting in a manipulated debt and borrow share.
+- **Insufficient Validation and Access Controls:** The lack of strict validation checks and access controls for custom spells and critical functions allowed unauthorized manipulation of the protocol.

```

Comment:
What 'custom spells' mean? I assume there is no such terminology in blockchain or smart contracts. Rephrase please